pytensor/tensor/rewriting/basic.py

```diff
@@ -975,6 +976,34 @@ def local_join_make_vector(fgraph, node):
         return [ret]
 
 
+@register_specialize
+@register_canonicalize
+@register_useless
+@node_rewriter([Sum])
+def local_sum_make_vector(fgraph, node):
+    """A sum of a MakeVector node is just the sum of the elements."""
+    (array,) = node.inputs
+
+    if array.owner is None:
+        return
+
+    if not isinstance(array.owner.op, MakeVector):
+        return
+
+    if node.op.axis == ():
+        return [array]
+
+    # If this is not the case the sum is invalid
+    assert node.op.axis is None or node.op.axis == (0,)
+
+    elements = array.owner.inputs
+    acc_dtype = node.op.acc_dtype
+    out_dtype = node.op.dtype
+    element_sum = cast(add(*[cast(value, acc_dtype) for value in elements]), out_dtype)
+
```

Comment:
Does `add` work with 0 / 1 inputs? I know MakeVector allows that

Reply:
I think this will work with one input, but fail with zero.

Reply:
True. I added a special case